lib/LocationSelection/LocationInput.js

```diff
@@ -50,12 +53,21 @@ export default class LocationInput extends React.Component {
   }
 
   render() {
-    const { resources, placeholder } = this.props;
+    const { resources, placeholder, remoteMap, intl } = this.props;
     const locations = (resources.locations || {}).records || [];
 
     if (!locations.length) return (<div />);
 
-    const locationOpts = locations.map(loc => ({ label: `${loc.name} (${loc.code})`, value: loc.id }));
+    const locationOpts = locations.map(loc => {
+      const remoteLocationLabel = (
+        remoteMap && (loc.id in remoteMap)
+      ) ? intl.formatMessage({ id: 'stripes-smart-components.remoteLabel' }) : '';
```

Comment:
IMHO, it's better to move it above and make one time call, use result after
`const remoteLabel = intl.formatMessage({ id: 'stripes-smart-components.remoteLabel' });`